clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -151,7 +151,7 @@ public Cluster buildCluster() {
             if (metadata.error == Errors.NONE) {
                 if (metadata.isInternal)
                     internalTopics.add(metadata.topic);
-                if (metadata.topicId() != null && metadata.topicId() != Uuid.ZERO_UUID) {
+                if (metadata.topicId() != null && !Uuid.ZERO_UUID.equals(metadata.topicId())) {
```

Comment:
Could we add a unit test to cover this one?

Reply:
As @jolshan say, MetadataResponseTest is less necessary since the server should be building the response, but doesn't hurt.

Reply:
I would add one to avoid a similar regression in the future.

Reply:
done. PTAL @dajac